bcda/auth/models_test.go

```diff
@@ -63,6 +66,81 @@ func (s *ModelsTestSuite) TestTokenCreation() {
 	assert.Equal(s.T(), tokenString, savedToken.TokenString)
 }
 
+func (s *ModelsTestSuite) TestGenerateSystemKeyPair() {
+	group := models.Group{GroupID: "abcdef123456"}
+	err := s.db.Create(&group).Error
+	if err != nil {
+		s.FailNow(err.Error())
+	}
+
+	system := models.System{
+		GroupID:  group.GroupID,
+		ClientID: "1234",
+	}
+	err = s.db.Create(&system).Error
+	if err != nil {
+		s.FailNow(err.Error())
+	}
+
+	privateKeyStr, err := auth.GenerateSystemKeyPair("1234")
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), privateKeyStr)
+
+	privKeyBlock, _ := pem.Decode([]byte(privateKeyStr))
+	privateKey, err := x509.ParsePKCS1PrivateKey(privKeyBlock.Bytes)
+	if err != nil {
+		s.FailNow(err.Error())
+	}
+
+	var pubEncrKey models.EncryptionKey
+	err = s.db.First(&pubEncrKey, "system_id = ?", system.ID).Error
+	if err != nil {
+		s.FailNow(err.Error())
+	}
+	pubKeyBlock, _ := pem.Decode([]byte(pubEncrKey.Body))
+	publicKey, err := x509.ParsePKIXPublicKey(pubKeyBlock.Bytes)
+	if err != nil {
+		s.FailNow(err.Error())
+	}
+	assert.Equal(s.T(), &privateKey.PublicKey, publicKey)
+
+	s.db.Unscoped().Delete(&pubEncrKey)
+	s.db.Unscoped().Delete(&system)
+	s.db.Unscoped().Delete(&group)
+}
+
+func (s *ModelsTestSuite) TestGenerateSystemKeyPair_AlreadyExists() {
```

Comment:
This is a great test of the expected database behavior.  Can/should we test that a valid key pair is generated?